### DISCUSSION TITLE
Unable to install in Yocto Scrathgap, missing setup.py

### DISCUSSION THREAD

**Discussion by mgarcia01752** [pygments/pygments#2701] [Q&A] [answered]
During the build, Pygements is missing setup.py.  Looking back, it was removed starting at 2.15

Not sure if it was an oversight. Can you verify and add to your package again?

`dev01@dev01:~/Projects/BareMetalRouterOS/poky/build-bmr$ bitbake python3-pygments
Loading cache: 100% |################################################################################################################################################################################| Time: 0:00:01
Loaded 4667 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "2.8.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "x86_64-poky-linux"
MACHINE              = "qemux86-64"
DISTRO               = "poky"
DISTRO_VERSION       = "5.0"
TUNE_FEATURES        = "m64 core2"
TARGET_FPU           = ""
meta                 
meta-poky            
meta-yocto-bsp       = "scarthgap:eb723d4a1082b10d14554f799c6c31cee620152d"
meta-oe              
meta-python          
meta-networking      = "scarthgap:a72010b414ee3d73888ac9cb4e310e8f05e13aea"
meta-intel           = "scarthgap:7864daf8ecec559fe144ba3d767768ba7392ec86"
meta-bare-metal-router = "scarthgap:eb723d4a1082b10d14554f799c6c31cee620152d"

Sstate summary: Wanted 10 Local 0 Mirrors 0 Missed 10 Current 602 (0% match, 98% complete)#########################################################################################                  | ETA:  0:00:00
Initialising tasks: 100% |###########################################################################################################################################################################| Time: 0:00:01
NOTE: Executing Tasks
ERROR: python3-pygments-2.17.2-r0 do_compile: 'python3 setup.py bdist_wheel ' execution failed.
ERROR: python3-pygments-2.17.2-r0 do_compile: ExecutionError('/home/dev01/Projects/BareMetalRouterOS/poky/build-bmr/tmp/work/core2-64-poky-linux/python3-pygments/2.17.2/temp/run.do_compile.2774454', 1, None, None)
ERROR: Logfile of failure stored in: /home/dev01/Projects/BareMetalRouterOS/poky/build-bmr/tmp/work/core2-64-poky-linux/python3-pygments/2.17.2/temp/log.do_compile.2774454
Log data follows:
| DEBUG: Executing shell function do_compile
| /home/dev01/Projects/BareMetalRouterOS/poky/build-bmr/tmp/work/core2-64-poky-linux/python3-pygments/2.17.2/recipe-sysroot-native/usr/bin/python3-native/python3: can't open file '/home/dev01/Projects/BareMetalRouterOS/poky/build-bmr/tmp/work/core2-64-poky-linux/python3-pygments/2.17.2/pygments-2.17.2/setup.py': [Errno 2] No such file or directory
| ERROR: 'python3 setup.py bdist_wheel ' execution failed.
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/dev01/Projects/BareMetalRouterOS/poky/meta-bare-metal-router/recipes-core/python3-jc/python3-pygments_2.17.2.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 1735 tasks of which 1734 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/dev01/Projects/BareMetalRouterOS/poky/meta-bare-metal-router/recipes-core/python3-jc/python3-pygments_2.17.2.bb:do_compile
Summary: There were 2 ERROR messages, returning a non-zero exit code.`

**Comment by jeanas** [comment, marked as answer]
This is intentional. Python packaging doesn't mandate `setup.py` files anymore. There's a [guide](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#what-commands-should-be-used-instead) on migrating direct `python setup.py` invocations.
